simplecpp.cpp

```diff
@@ -398,7 +398,7 @@ static unsigned short getAndSkipBOM(std::istream &istr)
 
     // The UTF-16 BOM is 0xfffe or 0xfeff.
     if (ch1 >= 0xfe) {
-        const unsigned short bom = (static_cast<unsigned char>(istr.get()) << 8);
+        const unsigned short bom = static_cast<unsigned short>((static_cast<unsigned char>(istr.get()) << 8));
```

Comment:
I don't see that this change improves the readability or makes the code more robust.

Reply:
It makes the implicit conversions explicit so you know what actually happens. If possible the code should be changed to avoid these. Since those might not be straight-forward I did not attempt this yet.

Reply:
I just don't agree. The code is longer and I don't see that it makes the code easier. On the contrary..

Reply:
The original code is shorter and 100% safe. I consider it a false positive from the compiler to warn about this. There can be no loss of precision.